crates/net/network/src/frost/manager.rs

```
@@ -0,0 +1,201 @@
+        }
+    }
+
+    fn all_peers_connected(&self) -> bool {
+        let peers_count = self.frost_peers_connections.keys().cloned().count();
+        peers_count == self.total_authorities - 1
+    }
+
+    /// Returns a new [`FrostHandle`] that can send commands to this type.
+    pub fn handle(&self) -> FrostHandle {
+        FrostHandle { manager_tx: self.command_tx.clone() }
+    }
+
+    fn on_network_event(&mut self, protocol_event: NetworkFrostEvent) {
+        match protocol_event {
+            NetworkFrostEvent::ConnectionEstablished {
+                direction,
+                peer_id,
+                to_connection,
+                authority_index,
+            } => {
+                info!(">>>>>>>>>>> FROST PEER CONNECTION ESTABLISHED");
+                // make sure we ignore our own connection
+                let my_peer_id = self.network.peer_id();
+                if *my_peer_id != peer_id {
+                    self.frost_peers_directions
+                        .insert(peer_id_to_identifier(authority_index), direction);
+                    self.frost_peers_connections
+                        .insert(peer_id_to_identifier(authority_index), to_connection);
+                }
+            }
+            NetworkFrostEvent::PeerMessage(response) => {
+                info!(">>>>>>>>>>> FROST PEER MESSAGE RECEIVED {:?}", response);
+                let _ = self.frost_task_forwarder_tx.send(response); // TODO:  handle error here
+            }
+        }
+    }
+
+    /// Handles a command received from a detached [`FrostHandle`]
+    fn on_command(&mut self, cmd: FrostCommand) {
+        match cmd {
+            FrostCommand::CheckConnectedToAll(tx) => {
+                // reply to caller
+                let _ = tx.send(self.all_peers_connected()).ok();
+            }
+            FrostCommand::GetAllConnectedFrostPeers(tx) => {
+                // reply to caller
+                let _ = tx.send(self.frost_peers_connections.clone()).ok();
```

nit: There `.ok()`'s are not having any need. If you want to ignore the error, you can just do `let _ = ` with the result without the `.ok()`.

---

agreed, missed that when coding :)